### DISCUSSION TITLE
Is it possible to run separate instances of postcss in the same project at the same time?

### DISCUSSION THREAD

**Discussion by nathangross** [postcss/postcss#1939] [Q&A]
I'm working in a Laravel (10) project using vite. Frontend we are using Tailwind 4 (alpha) and the backend (Filament PHP) requires Tailwind 3. Both are using postcss. 

Tailwind 3: postcss 8
Tailwind 4: @tailwind/postcss

I'm using npm workspaces to help manage the conflicting requirements. I have separate vite configs that point to separate postcss.config.js files in separate dirs.

```
{
    "name": "admin",
    "version": "1.0.0",
    "type": "module",
    "dependencies": {
        "@tailwindcss/forms": "^0.5.7",
        "@tailwindcss/typography": "^0.5.12",
        "autoprefixer": "^10.4.19",
        "postcss": "^8.4.38",
        "postcss-nesting": "^12.1.2",
        "tailwindcss": "^3.4.3"
    }
}
```

    
```
{
    "name": "frontend",
    "version": "1.0.0",
    "type": "module",
    "dependencies": {
        "@tailwindcss/postcss": "^4.0.0-alpha.14",
        "open-props": "^1.6.21",
        "tailwindcss": "^4.0.0-alpha.14"
    }
}
```

- When I run either single vite config (ex: `npm run dev:frontend`), both tailwind css versions are properly compiled by their respective versions ✅
- Also, I can run `npm run build:frontend & npm run build:admin` and they compile correctly ✅
- However, when I try to run both vite configs at the same time (`npm run dev:frontend & npm run dev:admin`) both tailwind css versions are compiled by tailwind 4 (so the tailwind 3 errors out) ❌ 

Is there a flag or config that I can set to ensure the postcss is handled separately when running `dev`?

Happy to provide more code, context, etc. if that helps. 

Thanks!

**Comment by ai** [comment]
Sorry, can’t help here because the conflict goes not from the PostCSS core, but from other environment

**Reply by nathangross** [reply]
Maybe if I ask in another way: is it possible to run multiple postcss "pipelines" at the same time—say in two different terminal instances?

**Reply by ai** [reply]
Yes. PostCSS core is just a process in memory. You can run as much as possible. There is nothing to have conflict with.

The problem is the environment around PostCSS core. Some PostCSS plugins can have temporary files, the bundlers and CLI pipeline tools often are very complex inside.
